dcp_prototype/backend/ledger/code/common/ledger_orm.py

```diff
@@ -24,161 +24,232 @@ def session(self, **kwargs):
-        "Library",
-        order_by=Library.id,
-        back_populates="project",
-        cascade="all, delete, delete-orphan",
-    )
+    id = Column(String, primary_key=True)
+    project_title = Column(String)
+    publication_title = Column(String)
+    publication_doi = Column(String)
+    external_accessions = Column(String)
+    created_at = Column(DateTime(True), nullable=False, server_default=text("now()"))
+    updated_at = Column(DateTime(True), nullable=False, server_default=text("now()"))
+
+
+class QuantificationProtocol(Base):
+    __tablename__ = "quantification_protocol"
+
+    id = Column(String, primary_key=True)
+    quantification_software = Column(String)
+    created_at = Column(DateTime(True), nullable=False, server_default=text("now()"))
+    updated_at = Column(DateTime(True), nullable=False, server_default=text("now()"))
 
 
 class SequenceFile(Base):
     __tablename__ = "sequence_file"
-    id = Column(String(), primary_key=True)
-    filename = Column(String(), nullable=True)
-    file_format = Column(String(), nullable=True)
-    flowcell_id = Column(String(), nullable=True)
-    lane_index = Column(String(), nullable=True)
-    read_index = Column(String(), nullable=True)
-    s3_uri = Column(String(), nullable=True)
-    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
-    updated_at = Column(
-        DateTime, default=datetime.utcnow, nullable=False, onupdate=datetime.utcnow
-    )
 
-    sequencing_protocol_id = Column(
-        String(), ForeignKey("sequencing_protocol.id"), nullable=False
-    )
-    sequencing_protocol = relationship(
-        "SequencingProtocol", back_populates="sequence_files"
-    )
+    id = Column(String, primary_key=True)
+    filename = Column(String)
+    file_format = Column(String)
+    file_size = Column(String)
+    flowcell_id = Column(String)
+    lane_index = Column(String)
+    read_index = Column(String)
+    s3_uri = Column(String, nullable=False)
+    created_at = Column(DateTime(True), nullable=False, server_default=text("now()"))
+    updated_at = Column(DateTime(True), nullable=False, server_default=text("now()"))
 
 
 class SequencingProtocol(Base):
     __tablename__ = "sequencing_protocol"
-    id = Column(String(), primary_key=True)
-    paired_end_sequencing = Column(String(), nullable=True)
-    instrument_manufacturer_model = Column(String(), nullable=True)
-    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
-    updated_at = Column(
-        DateTime, default=datetime.utcnow, nullable=False, onupdate=datetime.utcnow
+
+    id = Column(String, primary_key=True)
+    paired_end_sequencing = Column(String)
+    instrument_manufacturer_model = Column(String)
+    created_at = Column(DateTime(True), nullable=False, server_default=text("now()"))
+    updated_at = Column(DateTime(True), nullable=False, server_default=text("now()"))
+
+
+class AnalysisFileQuantificationProtocolExpressionFileProcessJ(Base):
```

Comment:
I think the last letter was supposed to be Join?

Reply:
fixed